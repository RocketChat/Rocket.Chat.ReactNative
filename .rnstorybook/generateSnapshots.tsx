import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
