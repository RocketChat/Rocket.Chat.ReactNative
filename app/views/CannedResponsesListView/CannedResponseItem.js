import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';

import Touchable from 'react-native-platform-touchable';
import { themes } from '../../constants/colors';
import * as List from '../../containers/List';
import styles from './styles';
import Button from '../../containers/Button';
import { CustomIcon } from '../../lib/Icons';
import I18n from '../../i18n';

const CannedResponseItem = ({
	theme, onPressDetail, shortcut, scope, onPressUse, text, tags
}) => (
	<>
		<Touchable onPress={onPressDetail} style={[styles.wrapCannedItem, { backgroundColor: themes[theme].messageboxBackground }]}>
			<>
				<View style={styles.cannedRow}>
					<View style={styles.cannedWrapShortcutScope}>
						<Text style={[styles.cannedShortcut, { color: themes[theme].titleText }]}>!{shortcut}</Text>
						<Text style={[styles.cannedScope, { color: themes[theme].auxiliaryTintColor }]}>{scope}</Text>
					</View>

					<Button
						title={I18n.t('Use')}
						fontSize={12}
						color={themes[theme].titleText}
						style={[styles.cannedUseButton, { backgroundColor: themes[theme].chatComponentBackground }]}
						theme={theme}
						onPress={onPressUse}
					/>

					<CustomIcon
						name='chevron-right'
						color={themes[theme].auxiliaryText}
						size={20}
					/>
				</View>

				<Text style={[styles.cannedText, { color: themes[theme].auxiliaryTintColor }]}>“{text}”</Text>

				<View style={styles.cannedTagContainer}>
					{
						tags?.length > 0
							? tags.map(t => (
								<View style={[styles.cannedTagWrap, { backgroundColor: themes[theme].searchboxBackground }]}>
									<Text style={[styles.cannedTag, { color: themes[theme].auxiliaryTintColor }]}>{t}</Text>
								</View>
							))
							: null
					}
				</View>
			</>
		</Touchable>
		<List.Separator />
	</>

);

CannedResponseItem.propTypes = {
	theme: PropTypes.string,
	onPressDetail: PropTypes.func,
	shortcut: PropTypes.string,
	scope: PropTypes.string,
	onPressUse: PropTypes.func,
	text: PropTypes.string,
	tags: PropTypes.array
};

CannedResponseItem.defaultProps = {
	onPressDetail: () => {},
	onPressUse: () => {}
};

export default CannedResponseItem;
