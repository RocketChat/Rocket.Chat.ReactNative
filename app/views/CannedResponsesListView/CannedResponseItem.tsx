import React from 'react';
import { View, Text } from 'react-native';
import Touchable from 'react-native-platform-touchable';

import { TSupportedThemes } from '../../theme';
import { themes } from '../../lib/constants';
import Button from '../../containers/Button';
import I18n from '../../i18n';
import styles from './styles';

interface ICannedResponseItem {
	theme: TSupportedThemes;
	onPressDetail: () => void;
	shortcut: string;
	scope: string;
	onPressUse: () => void;
	text: string;
	tags?: string[];
}

const CannedResponseItem = ({
	theme,
	onPressDetail = () => {},
	shortcut,
	scope,
	onPressUse = () => {},
	text,
	tags = []
}: ICannedResponseItem): JSX.Element => (
	<Touchable onPress={onPressDetail} style={[styles.wrapCannedItem, { backgroundColor: themes[theme].messageboxBackground }]}>
		<>
			<View style={styles.cannedRow}>
				<View style={styles.cannedWrapShortcutScope}>
					<Text style={[styles.cannedShortcut, { color: themes[theme].titleText }]}>!{shortcut}</Text>
					<Text style={[styles.cannedScope, { color: themes[theme].auxiliaryTintColor }]}>{scope}</Text>
				</View>

				<Button
					title={I18n.t('Use')}
					fontSize={12}
					color={themes[theme].titleText}
					style={[styles.cannedUseButton, { backgroundColor: themes[theme].chatComponentBackground }]}
					onPress={onPressUse}
				/>
			</View>

			<Text ellipsizeMode='tail' numberOfLines={2} style={[styles.cannedText, { color: themes[theme].auxiliaryTintColor }]}>
				“{text}”
			</Text>
			<View style={styles.cannedTagContainer}>
				{tags?.length > 0
					? tags.map(t => (
							<View style={[styles.cannedTagWrap, { backgroundColor: themes[theme].searchboxBackground }]}>
								<Text style={[styles.cannedTag, { color: themes[theme].auxiliaryTintColor }]}>{t}</Text>
							</View>
					  ))
					: null}
			</View>
		</>
	</Touchable>
);

export default CannedResponseItem;
