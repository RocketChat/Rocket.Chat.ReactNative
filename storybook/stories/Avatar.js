import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import { themes } from '../../app/constants/colors';
import Avatar from '../../app/containers/Avatar/Avatar';
import Status from '../../app/containers/Status/Status';
import StoriesSeparator from './StoriesSeparator';
import sharedStyles from '../../app/views/Styles';

const styles = StyleSheet.create({
	status: {
		borderWidth: 4,
		bottom: -4,
		right: -4
	},
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

const Separator = ({ title, theme }) => <StoriesSeparator title={title} theme={theme} />;
Separator.propTypes = {
	title: PropTypes.string,
	theme: PropTypes.string
};

const AvatarStories = ({ theme }) => (
	<ScrollView style={{ backgroundColor: themes[theme].backgroundColor }}>
		<Separator title='Avatar by text' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by roomId' theme={theme} />
		<Avatar
			type='p'
			rid='devWBbYr7inwupPqK'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by url' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			size={56}
		/>
		<Separator title='Avatar by path' theme={theme} />
		<Avatar
			avatar='/avatar/diego.mello'
			server={server}
			size={56}
		/>
		<Separator title='With ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			avatarETag='5ag8KffJcZj9m5rCv'
			server={server}
			size={56}
		/>
		<Separator title='Without ETag' theme={theme} />
		<Avatar
			type='d'
			text='djorkaeff.alexandre'
			server={server}
			size={56}
		/>
		<Separator title='Emoji' theme={theme} />
		<Avatar
			emoji='troll'
			getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })}
			server={server}
			size={56}
		/>
		<Separator title='Direct' theme={theme} />
		<Avatar
			text='diego.mello'
			server={server}
			type='d'
			size={56}
		/>
		<Separator title='Channel' theme={theme} />
		<Avatar
			text='general'
			server={server}
			type='c'
			size={56}
		/>
		<Separator title='Touchable' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			onPress={() => console.log('Pressed!')}
			size={56}
		/>
		<Separator title='Static' theme={theme} />
		<Avatar
			avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
			server={server}
			isStatic
			size={56}
		/>
		<Separator title='Custom borderRadius' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			borderRadius={28}
			size={56}
		/>
		<Separator title='Children' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
		>
			<Status
				size={24}
				style={[sharedStyles.status, styles.status]}
				theme={theme}
			/>
		</Avatar>
		<Separator title='Wrong server' theme={theme} />
		<Avatar
			text='Avatar'
			server='https://google.com'
			size={56}
		/>
		<Separator title='Custom style' theme={theme} />
		<Avatar
			text='Avatar'
			server={server}
			size={56}
			style={styles.custom}
		/>
	</ScrollView>
);
AvatarStories.propTypes = {
	theme: PropTypes.string
};
export default AvatarStories;
