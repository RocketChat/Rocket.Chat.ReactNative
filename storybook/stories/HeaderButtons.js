/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions, react/prop-types */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import * as HeaderButton from '../../app/containers/HeaderButton';
import Header from '../../app/containers/Header';
import { ThemeContext } from '../../app/theme';

const stories = storiesOf('Header Buttons', module);

const HeaderExample = ({ left, right }) => (
	<Header
		headerLeft={left}
		headerTitle={() => <View style={{ flex: 1 }} />}
		headerRight={right}
	/>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }) => (
	<ThemeContext.Provider
		value={{ theme }}
	>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Drawer />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CloseModal />
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.CancelModal />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.More />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Download />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Preferences />
			)}
		/>
		<HeaderExample
			right={() => (
				<HeaderButton.Legal />
			)}
		/>
	</>
));
