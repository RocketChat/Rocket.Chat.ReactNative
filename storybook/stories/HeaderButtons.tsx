import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { SafeAreaView } from 'react-native';
import { Header, HeaderBackground } from '@react-navigation/elements';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import * as HeaderButton from '../../app/containers/HeaderButton';
import { TColors, ThemeContext, TSupportedThemes } from '../../app/theme';
import { colors } from '../../app/lib/constants';

const stories = storiesOf('Header Buttons', module).addDecorator(story => <SafeAreaProvider>{story()}</SafeAreaProvider>);

interface IHeader {
	left?: () => React.ReactElement | null;
	right?: () => React.ReactElement;
	title?: string;
	colors?: TColors;
}

const HeaderExample = ({ left, right, colors, title = '' }: IHeader) => (
	<SafeAreaView>
		<Header
			title={title}
			headerLeft={left}
			headerRight={right}
			headerBackground={() => <HeaderBackground style={{ backgroundColor: colors?.headerBackground }} />}
		/>
	</SafeAreaView>
);

stories.add('title', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='threads' />
					<HeaderButton.Item title='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('icons', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
		/>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item iconName='threads' />
					<HeaderButton.Item iconName='search' />
				</HeaderButton.Container>
			)}
		/>
	</>
));

stories.add('badge', () => (
	<>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadUser={[1]} />} />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} tunreadGroup={[1]} />} />
				</HeaderButton.Container>
			)}
		/>
	</>
));

const ThemeStory = ({ theme }: { theme: TSupportedThemes }) => (
	<ThemeContext.Provider value={{ theme, colors: colors[theme] }}>
		<HeaderExample
			left={() => (
				<HeaderButton.Container left>
					<HeaderButton.Item iconName='threads' />
				</HeaderButton.Container>
			)}
			right={() => (
				<HeaderButton.Container>
					<HeaderButton.Item title='Threads' />
					<HeaderButton.Item iconName='threads' badge={() => <HeaderButton.Badge tunread={[1]} />} />
				</HeaderButton.Container>
			)}
			colors={colors[theme]}
		/>
	</ThemeContext.Provider>
);

stories.add('themes', () => (
	<>
		<ThemeStory theme='light' />
		<ThemeStory theme='dark' />
		<ThemeStory theme='black' />
	</>
));

stories.add('common', () => (
	<>
		<HeaderExample left={() => <HeaderButton.Drawer />} />
		<HeaderExample left={() => <HeaderButton.CloseModal />} />
		<HeaderExample left={() => <HeaderButton.CancelModal />} />
		<HeaderExample right={() => <HeaderButton.More />} />
		<HeaderExample right={() => <HeaderButton.Download />} />
		<HeaderExample right={() => <HeaderButton.Preferences />} />
		<HeaderExample right={() => <HeaderButton.Legal />} />
	</>
));
