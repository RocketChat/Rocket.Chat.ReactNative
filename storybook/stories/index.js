/* eslint-disable import/no-extraneous-dependencies, import/no-unresolved, import/extensions */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { storiesOf } from '@storybook/react-native';

import RoomItem from './RoomItem';
import Message from './Message';
import UiKitMessage from './UiKitMessage';
import UiKitModal from './UiKitModal';
import Markdown from './Markdown';
// import RoomViewHeader from './RoomViewHeader';

import MessageContext from '../../app/containers/message/Context';

// MessageProvider
const baseUrl = 'https://open.rocket.chat';
const user = {
	id: '',
	username: 'diego.mello',
	token: ''
};

// Change here to see themed storybook
const theme = 'light';

const reducers = combineReducers({
	settings: () => ({}),
	login: () => ({
		user: {
			username: 'diego.mello'
		}
	}),
	meteor: () => ({ connected: true }),
	activeUsers: () => ({ abc: { status: 'online', statusText: 'dog' } })
});
const store = createStore(reducers);

const messageDecorator = story => (
	<MessageContext.Provider
		value={{
			user,
			baseUrl,
			onPress: () => {},
			onLongPress: () => {},
			reactionInit: () => {},
			onErrorPress: () => {},
			replyBroadcast: () => {},
			onReactionPress: () => {},
			onDiscussionPress: () => {},
			onReactionLongPress: () => {}
		}}
	>
		{story()}
	</MessageContext.Provider>
);

storiesOf('RoomItem', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.add('list roomitem', () => <RoomItem theme={theme} />);
storiesOf('Message', module)
	.addDecorator(messageDecorator)
	.add('list message', () => <Message theme={theme} />);

storiesOf('UiKitMessage', module)
	.addDecorator(messageDecorator)
	.add('list uikitmessage', () => <UiKitMessage theme={theme} />);
storiesOf('UiKitModal', module)
	.addDecorator(messageDecorator)
	.add('list UiKitModal', () => <UiKitModal theme={theme} />);
storiesOf('Markdown', module)
	.add('list Markdown', () => <Markdown theme={theme} />);

// FIXME: I couldn't make these pass on jest :(
// storiesOf('RoomViewHeader', module)
// 	.add('list', () => <RoomViewHeader theme='black' />);
