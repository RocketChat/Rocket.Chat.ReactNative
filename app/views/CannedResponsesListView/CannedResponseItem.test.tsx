import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './CannedResponseItem.stories';

generateSnapshots(stories);
