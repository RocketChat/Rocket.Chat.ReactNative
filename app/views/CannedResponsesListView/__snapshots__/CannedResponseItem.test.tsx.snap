// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Itens should match snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "maxHeight": 141,
        "minHeight": 117,
        "opacity": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "height": 36,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontFamily": "Inter",
                "fontSize": 14,
                "fontWeight": "500",
                "paddingBottom": 0,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          !
          !FAQ4
        </Text>
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Private
        </Text>
      </View>
      <View
        accessibilityLabel="Use"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#F7F8FA",
            "borderRadius": 4,
            "height": 28,
            "justifyContent": "center",
            "marginBottom": 12,
            "marginLeft": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 14,
            "width": 56,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#1F2329",
                "fontSize": 12,
              },
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Use
        </Text>
      </View>
    </View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 14,
            "fontWeight": "400",
            "marginTop": 8,
            "paddingBottom": 0,
            "paddingTop": 0,
            "textAlign": "left",
          },
          {
            "color": "#6C727A",
          },
        ]
      }
    >
      “
      ZCVXZVXCZVZXVZXCVZXCVXZCVZX
      ”
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    />
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "maxHeight": 141,
        "minHeight": 117,
        "opacity": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "height": 36,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontFamily": "Inter",
                "fontSize": 14,
                "fontWeight": "500",
                "paddingBottom": 0,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          !
          test4mobilePrivate
        </Text>
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Private
        </Text>
      </View>
      <View
        accessibilityLabel="Use"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#F7F8FA",
            "borderRadius": 4,
            "height": 28,
            "justifyContent": "center",
            "marginBottom": 12,
            "marginLeft": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 14,
            "width": 56,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#1F2329",
                "fontSize": 12,
              },
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "500",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Use
        </Text>
      </View>
    </View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 14,
            "fontWeight": "400",
            "marginTop": 8,
            "paddingBottom": 0,
            "paddingTop": 0,
            "textAlign": "left",
          },
          {
            "color": "#6C727A",
          },
        ]
      }
    >
      “
      test for mobile private
      ”
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          HQ
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Closed
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          HQ
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Problem in Product Y
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          HQ
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Closed
        </Text>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 4,
              "height": 16,
              "marginRight": 4,
              "marginTop": 8,
            },
            {
              "backgroundColor": "#EBECEF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 12,
                "fontWeight": "400",
                "paddingBottom": 0,
                "paddingHorizontal": 4,
                "paddingTop": 0,
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          Problem in Product Y
        </Text>
      </View>
    </View>
  </View>,
]
`;
